bot/seasons/evergreen/minesweeper.py

```diff
@@ -0,0 +1,198 @@
+import typing
+from random import random
+
+import discord
+from discord.ext import commands
+
+
+class Minesweeper(commands.Cog):
+    """Play a game of minesweeper."""
+
+    def __init__(self, bot: commands.Bot) -> None:
+        self.bot = bot
+        self.games: typing.Dict[discord.member, typing.Dict] = {}  # Store the currently running games
+
+    @staticmethod
+    def is_bomb(cell: typing.Union[str, int]) -> int:
+        """Returns 1 if `cell` is a bomb if not 0"""
+        return 1 if cell == "bomb" else 0
+
+    def generate_board(self, bomb_chance: float) -> typing.List[typing.List[typing.Union[str, int]]]:
+        """Generate a 2d array for the board."""
+        board: typing.List[typing.List[typing.Union[str, int]]] = [
+            ["bomb" if random() <= bomb_chance else "number" for _ in range(10)] for _ in range(10)]
+        for y, row in enumerate(board):
+            for x, cell in enumerate(row):
+                if cell == "number":
+                    # calculate bombs near it
+                    to_check = []
+                    for xt in [x - 1, x, x + 1]:
+                        for yt in [y - 1, y, y + 1]:
+                            if xt != -1 and xt != 10 and yt != -1 and yt != 10:
+                                to_check.append(board[yt][xt])
+
+                    bombs = sum(map(self.is_bomb, to_check))
```

Comment:
Personally prefer a generator / list comprehension here as `sum(self.is_bomb(cell) for cell in to_check)` or better yet just check the cell in the loop to prevent 2 iteration?  You don't use `to_check` anywhere else that i can see, maybe a `collections.Counter`?